presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/IterativeOptimizer.java

```diff
@@ -76,25 +79,27 @@ public PlanNode optimize(PlanNode plan, Session session, Map<Symbol, Type> types
 
         Memo memo = new Memo(idAllocator, plan);
         Lookup lookup = Lookup.from(memo::resolve);
+        Matcher matcher = DEFAULT_MATCHER;
```

Comment:
When would the matcher not be DEFAULT_MATCHER? Also, since it seems to be constant for the class make it a static or instance field instead of passing it around.